test/ess.jl

```diff
@@ -1,55 +1,94 @@
-        @test reshape(y, size(x)) == x
-
-        # check a matrix with odd number of rows
-        x = rand(51, 20)
-
-        # check incompatible sizes
-        @test_throws DimensionMismatch MCMCDiagnosticTools.copyto_split!(
-            similar(x, 25, 20), x
-        )
-        @test_throws DimensionMismatch MCMCDiagnosticTools.copyto_split!(
-            similar(x, 51, 40), x
-        )
-
-        MCMCDiagnosticTools.copyto_split!(y, x)
-        @test reshape(y, 50, 20) == x[vcat(1:25, 27:51), :]
+using Distributions
+using DynamicHMC
+using LogDensityProblems
+using LogExpFunctions
+using MCMCDiagnosticTools
+using MCMCDiagnosticTools: _rank_normalize
+using Random
+using Statistics
+using StatsBase
+using Test
+
+struct CauchyProblem end
+LogDensityProblems.logdensity(p::CauchyProblem, θ) = -sum(log1psq, θ)
+function LogDensityProblems.logdensity_and_gradient(p::CauchyProblem, θ)
+    return -sum(log1psq, θ), -2 .* θ ./ (1 .+ θ .^ 2)
+end
+LogDensityProblems.dimension(p::CauchyProblem) = 50
+function LogDensityProblems.capabilities(p::CauchyProblem)
+    return LogDensityProblems.LogDensityOrder{1}()
+end
+
+# AR(1) process
+function ar1(rng::AbstractRNG, φ::Real, σ::Real, n::Int...)
+    T = float(Base.promote_eltype(φ, σ))
+    x = randn(rng, T, n...)
+    x .*= σ
+    accumulate!(x, x; dims=1) do xi, ϵ
+        return muladd(φ, xi, ϵ)
     end
+    return x
+end
 
-    @testset "ESS and R̂ (IID samples)" begin
-        rawx = randn(10_000, 10, 40)
+asymptotic_dist(::typeof(mean), dist) = Normal(mean(dist), std(dist))
+function asymptotic_dist(::typeof(var), dist)
+    μ = var(dist)
+    σ = μ * sqrt(kurtosis(dist) + 2)
+    return Normal(μ, σ)
+end
+function asymptotic_dist(::typeof(std), dist)
+    μ = std(dist)
+    σ = μ * sqrt(kurtosis(dist) + 2) / 2
+    return Normal(μ, σ)
+end
+asymptotic_dist(::typeof(median), dist) = asymptotic_dist(Base.Fix2(quantile, 1//2), dist)
+function asymptotic_dist(f::Base.Fix2{typeof(quantile),<:Real}, dist)
+    p = f.x
+    μ = quantile(dist, p)
+    σ = sqrt(p * (1 - p)) / pdf(dist, μ)
+    return Normal(μ, σ)
+end
+function asymptotic_dist(::typeof(mad), dist::Normal)
+    # Example 21.10 of Asymptotic Statistics. Van der Vaart
+    d = Normal(zero(dist.μ), dist.σ)
+    dtrunc = truncated(d; lower=0)
+    μ = median(dtrunc)
+    σ = 1 / (4 * pdf(d, quantile(d, 3//4)))
+    return Normal(μ, σ) / quantile(Normal(), 3//4)
+end
 
+@testset "ess.jl" begin
+    @testset "ESS and R̂ (IID samples)" begin
         # Repeat tests with different scales
-        for scale in (1, 50, 100)
-            x = scale * rawx
-
-            ess_standard, rhat_standard = ess_rhat(x)
-            ess_standard2, rhat_standard2 = ess_rhat(x; method=ESSMethod())
-            ess_fft, rhat_fft = ess_rhat(x; method=FFTESSMethod())
-            ess_bda, rhat_bda = ess_rhat(x; method=BDAESSMethod())
+        @testset "scale=$scale, nchains=$nchains, split_chains=$split_chains" for scale in
```

Comment:
This is the default description/name anyway IIRC.
```suggestion
        @testset for scale in
```